pkg/operator/genericoperatorclient/dynamic_operator_client.go

```diff
@@ -156,6 +157,68 @@ func (c dynamicOperatorClient) UpdateOperatorStatus(resourceVersion string, stat
 	return retStatus, nil
 }
 
+func (c dynamicOperatorClient) EnsureFinalizer(finalizer string) error {
+	uncastInstance, err := c.informer.Lister().Get(c.configName)
+	if err != nil {
+		return err
+	}
+
+	instance := uncastInstance.(*unstructured.Unstructured)
+	finalizers := instance.GetFinalizers()
+	for _, f := range finalizers {
+		if f == finalizer {
+			return nil
+		}
+	}
+
+	// Change is needed
+	klog.V(4).Infof("Adding finalizer %q", finalizer)
+	newFinalizers := append(finalizers, finalizer)
+	err = c.saveFinalizers(instance, newFinalizers)
+	if err != nil {
+		return err
+	}
+	klog.V(2).Infof("Added finalizer %q", finalizer)
+	return err
+}
+
+func (c dynamicOperatorClient) RemoveFinalizer(finalizer string) error {
+	uncastInstance, err := c.informer.Lister().Get(c.configName)
+	if err != nil {
+		return err
+	}
+
+	instance := uncastInstance.(*unstructured.Unstructured)
+	finalizers := instance.GetFinalizers()
+	found := false
+	newFinalizers := make([]string, 0, len(finalizers))
+	for _, f := range finalizers {
+		if f == finalizer {
+			found = true
+			continue
+		}
+		newFinalizers = append(newFinalizers, f)
+	}
+	if !found {
+		return nil
+	}
+
+	klog.V(4).Infof("Removing finalizer %q: %v", finalizer, newFinalizers)
+	err = c.saveFinalizers(instance, newFinalizers)
+	if err != nil {
+		return err
+	}
+	klog.V(2).Infof("Removed finalizer %q", finalizer)
+	return nil
+}
+
+func (c dynamicOperatorClient) saveFinalizers(instance *unstructured.Unstructured, finalizers []string) error {
+	clone := instance.DeepCopy()
+	clone.SetFinalizers(finalizers)
+	_, err := c.client.Update(context.TODO(), clone, metav1.UpdateOptions{})
```

Comment:
this doesn't look right. We have an UpdateStatus that takes a func to apply a mutation against conflicts.  I think you should do the same here.

Reply:
All other functions of `dynamicOperatorClient` do not retry on conflict, adding retry here would be odd. I added retry funcs to v1helpers, following `UpdateStatus` pattern.

Reply:
> All other functions of `dynamicOperatorClient` do not retry on conflict, adding retry here would be odd. I added retry funcs to v1helpers, following `UpdateStatus` pattern.

sounds good.